ow_power_system/src/PowerSystemPack.cpp

```diff
@@ -0,0 +1,375 @@
+// TODO (JW): By its nature, this file is designed to be copied. It should
+// be explicitely (un)licensed as public domain or CC0.
+#include <chrono>
+#include <ctime>
+#include <fstream>
+#include <iomanip>
+#include <map>
+#include <sstream>
+#include <vector>
+#include <numeric>
+#include <math.h>
+#include <algorithm>
+#include <ros/package.h>
+#include <std_msgs/Float64.h>
+#include <std_msgs/Int16.h>
+#include <ow_lander/lander_joints.h>
+
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include <PrognoserFactory.h>
+
+#include "Messages/MessageBus.h"
+#include "Messages/ProgEventMessage.h"
+#include "Messages/ScalarMessage.h"
+#include "ModelBasedAsyncPrognoserBuilder.h"
+
+#include "PowerSystemPack.h"
+#include "PowerSystemNode.h"
+
+using namespace PCOE;
+
+const auto START_TIME       = MessageClock::now();
+//const int NUM_NODES         = 8; // NOTE: Should equal m_num_nodes in PowerSystemPack.h
+
+// The index use to access temperature information.
+// This might change to median SOC or RUL index or fixed percentile.
+//
+static constexpr int TEMPERATURE_INDEX = 1;
+
+double EoD_events[NUM_NODES][3];
+
+// The PredictionHandler class subscribes to the battery EoD event message and
+// prints each event as it is received.
+class PredictionHandler : public IMessageProcessor 
```

Comment:
I think moving it to its own source & header file is ideal, though at the moment it uses a global variable within PowerSystemPack.cpp (EoD_events[][]) to share the EoD predictions with the PowerSystemPack class. I don't like this current solution, and I do think cleaning it up and splitting the classes into 2 files would be better, but at the moment it would interfere significantly with the implementation. Maybe I could post this change as a JIRA ticket later down the line?